include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -775,6 +810,35 @@ class SharedDataFacade final : public BaseDataFacade
         }
     }
 
+    // Returns the data source ids that were used to supply the edge
+    // weights.
+    virtual void
+    GetUncompressedReverseDatasources(const EdgeID id,
+                                      std::vector<uint8_t> &result_datasources) const override final
+    {
+        const unsigned begin = m_geometry_indices.at(id);
+        const unsigned end = m_geometry_indices.at(id + 1) - 1;
+
+        result_datasources.clear();
+        result_datasources.reserve(end - begin);
+
+        // If there was no datasource info, return an array of 0's.
+        if (m_datasource_list.empty())
+        {
+            for (unsigned i = 0; i < end - begin; ++i)
+            {
+                result_datasources.push_back(0);
+            }
+        }
+        else
+        {
+            std::for_each(
```

Comment:
This is the goal of https://github.com/Project-OSRM/osrm-backend/issues/2575

It might be better to have this as a separate PR.